### PR TITLE
Rework the use of ParameterList in THCM.

### PR DESCRIPTION
**Pull request by merijn**
Adds getDefaultParameters and getDefaultInitParameters for OMUSE to be able to query the parameters it can use. Also introduces validation of the parameters passed in.

Recreated due to my failed GitHub-Fu.

### CODE REVIEW

File: src/ocean/THCM.C [hunk 1]
```diff
@@ -828,6 +830,12 @@ THCM::THCM(Teuchos::ParameterList& params, Teuchos::RCP<Epetra_Comm> comm) :
         col_scaling->PutScalar(1.0);
     }
 
+    for (auto& pair : paramList.sublist("Starting Parameters")) {
```
Comment (Sbte) [inline src/ocean/THCM.C]: Why is this suddenly needed? If it is not yet needed, please move it to another commit. Also, why not call `setParameters`?

Comment (Sbte) [reply src/ocean/THCM.C]: Sorry, I guess this is to make the parameters used. It changes the behaviour, however, so I'd still move it to a separate commit.

Comment (merijn) [reply src/ocean/THCM.C]: Moved to a different commit
